src/nvim/eval.lua

```diff
@@ -17,9 +17,9 @@
 --- @field deprecated? true
 --- @field returns? string|false
 --- @field returns_desc? string
---- @field signature string
+--- @field signature? string
 --- @field desc? string
---- @field params {[1]:string, [2]:string, [3]:string}[]
+--- @field params? {[1]:string, [2]:string, [3]:string}[]
```

Comment:
I intentionally left these as required in order to resolve the dagnostics some other way.

Reply:
Well, then these need to be provided for the commands that are missing them (`foreground`, `test_garbagecollect_now`, `test_write_list_log`).

Reply:
Yep, none of these are documented. I'm not sure we even need `foreground()` as it's a no-op. And the `test_*` functions just need a signature/params provided.

Alternatively we could make `signature` optional and derive it from params. Either way I think `params` needs to be required.